a10_octavia/controller/worker/tasks/glm_tasks.py

```diff
@@ -188,3 +188,46 @@ def execute(self, vthunder):
             LOG.warning("No vthunder therefore license revocation cannot occur.")
             return None
         self.axapi_client.delete.glm_license.post()
+
+
+class ConfigureForwardProxyServer(task.Task):
+    def __init__(self, *args, **kwargs):
+        self.host = self.port = self.username \
+            = self.enable_password = self.secret_string = None
+        super(ConfigureForwardProxyServer, self).__init__(*args, **kwargs)
+
+    def fetch_configurations(self, flavor):
```

Comment:
We should keep the precedence approach to flavor vs config instead of the all-or-none approach defined here. The issue is that the user may have a single proxy host where they have multiple proxies configured on different ports. Or perhaps they want some devices to login as one user and others to login as other users.


```
proxy_fl = flavor.get('proxy-server') if flavor and flavor.get('proxy-server') else {}
self.host = proxy_fl.get('proxy_host') if proxy_fl.get('proxy_host') else CONF.glm_license.proxy_host
......
```

This gives the user more freedom

Reply:
Actually found an even better way to do this since facing same issue on GLM activation.

```
def build_configuration(self, flavor):
        glm_config = dict(CONF.glm_license)
        glm_config.update(flavor.get('proxy-server'))
        
       config_payload = {
           'proxy-server': glm_config.get('proxy-server'),
           .....
       }
      return config_payload

def execute(...):
    ....
    config_payload = build_configuration(flavor)
    ....
    api_call(arg1, arg2, **config_payload)
```

Reply:
done